packages/hawtio/src/plugins/camel/route-diagram/RouteDiagram.tsx

```diff
@@ -163,8 +178,27 @@ const ReactFlowRouteDiagram: React.FunctionComponent<ReactFlowRouteDiagramProps>
     }
   }, [statsXml, setNodes, nodes.length])
 
+  const hasCause = (error: Error) => {
+    if (!error || !error.cause) return false
+    return error.cause instanceof Error
+  }
+
+  if (error) {
+    return (
+      <Card>
+        <CardBody>
+          <Alert variant='warning' title='Route could not be drawn.' />
+
+          <Alert variant='danger' title={error.message}>
+            {hasCause(error) && <p>Cause: {(error.cause as Error).message}</p>}
+          </Alert>
+        </CardBody>
+      </Card>
+    )
+  }
```

Comment:
why can't we use the existing `eventService` function to notify errors instead?

Reply:
Yeah, as far as I remember there are no other places that use a card for error. For consistentcy it's better to use the `eventService` for error notification. We can still keep the warning message if a blank area isn't comfortable, or there's indeed a dedicated component for the empty state.
https://v5-archive.patternfly.org/components/empty-state